tests/check_warnings.py

```diff
@@ -0,0 +1,56 @@
+from pathlib import Path
+import sys
+
+from colorama import Fore, init
+
+# init colors for all plateforms
+init()
+
+
+def check_warnings(file):
+    """
+    Check the list of warnings produced by the GitHub CI tests
+    raise errors if there are unexpected ones and/or if some are missing
+
+    Args:
+        file (pathlib.Path): the path to the generated warning.txt file from
+            the CI build
+
+    Return:
+        0 if the warnings are all there
+        1 if some warning are not registered or unexpected
+    """
+
+    # print some log
+    print("\n=== Sphinx Warnings test ===\n\n")
+
+    # find the file where all the known warnings are stored
+    warning_file = Path(__file__).parent / "warning_list.txt"
```

Comment:
this makes it seem like we are hard-coding the name `warning_list.txt` even if the provided path was a different file name like `warning.txt`...is that intentional?

Reply:
`warning_list` is the test file and `warning.txt` is the tested file. the test file is intended to be hard-coded as this is where I store the warnings

Reply:
ah I see 👍